python/dgl/distributed/graph_partition_book.py

```diff
@@ -237,3 +226,95 @@ def get_partition(self, partid):
                 getting remote partitions.')
 
         return self._graph
+
+class PartitionPolicy(object):
+    """Wrapper for GraphPartitionBook and RangePartitionBook.
+
+    We can extend this class to support HeteroGraph in the future.
+
+    Parameters
+    ----------
+    policy_str : str
+        partition-policy string, e.g., 'edge' or 'node'.
+    part_id : int
+        partition ID
+    partition_book : GraphPartitionBook or RangePartitionBook
+        Main class storing the partition information
+    """
+    def __init__(self, policy_str, part_id, partition_book):
+        # TODO(chao): support more policies for HeteroGraph
+        assert policy_str in ('edge', 'node'), 'policy_str must be \'edge\' or \'node\'.'
+        assert part_id >= 0, 'part_id %d cannot be a negative number.' % part_id
+        self._policy_str = policy_str
+        self._part_id = part_id
+        self._partition_book = partition_book
+
+    @property
+    def policy_str(self):
+        """Get policy string"""
+        return self._policy_str
+
+    @property
+    def part_id(self):
+        """Get partition ID"""
+        return self._part_id
+
+    @property
+    def partition_book(self):
+        """Get partition book"""
+        return self._partition_book
+
+    def to_local(self, id_tensor):
+        """Mapping global ID to local ID.
+
+        Parameters
+        ----------
+        id_tensor : tensor
+            Gloabl ID tensor
+
+        Return
+        ------
+        tensor
+            local ID tensor
+        """
+        if self._policy_str == 'edge':
+            return self._partition_book.eid2localeid(id_tensor, self._part_id)
+        elif self._policy_str == 'node':
+            return self._partition_book.nid2localnid(id_tensor, self._part_id)
+        else:
+            raise RuntimeError('Cannot support policy: %s ' % self._policy_str)
+
+    def to_partid(self, id_tensor):
+        """Mapping global ID to partition ID.
+
+        Parameters
+        ----------
+        id_tensor : tensor
+            Global ID tensor
+
+        Return
+        ------
+        tensor
+            partition ID
+        """
+        if self._policy_str == 'edge':
+            return self._partition_book.eid2partid(id_tensor)
+        elif self._policy_str == 'node':
+            return self._partition_book.nid2partid(id_tensor)
+        else:
+            raise RuntimeError('Cannot support policy: %s ' % self._policy_str)
+
+    def get_data_size(self):
+        """Get data size of current partition.
+
+        Returns
+        -------
+        int
+            data size
+        """
+        if self._policy_str == 'edge':
+            return len(self._partition_book.partid2eids(self._part_id))
+        elif self._policy_str == 'node':
+            return len(self._partition_book.partid2nids(self._part_id))
```

Comment:
this operation might be expensive. it's better to add an API to get data size in a more efficient way.